bootstrap.sh

```diff
@@ -1,10 +1,77 @@
 #!/bin/bash
 
-set -ueo pipefail
-
 echo "/dev/disk/by-id/google-elasticpd /var/lib/elasticsearch ext4 defaults 0 0" >> /etc/fstab
 lsblk -o label /dev/disk/by-id/google-elasticpd | grep -q elastic-data || mkfs.ext4 -L elastic-data /dev/disk/by-id/google-elasticpd
 mount /var/lib/elasticsearch
 chown elasticsearch:elasticsearch /var/lib/elasticsearch
 
+
+if ! grep 'elasticsearch soft memlock unlimited' /etc/security/limits.conf; then
+
+(
+cat <<ADDTEXT
+elasticsearch soft memlock unlimited
+elasticsearch hard memlock unlimited
+ADDTEXT
+) >> /etc/security/limits.conf;
+
+fi
+
+su elasticsearch -s /bin/bash -c '/usr/share/elasticsearch/bin/elasticsearch | tee /tmp/elasticsearch_bootstrap_start.log >/dev/full' &
+
+echo "waiting for ES7 bootstrap to start"
+while true; do
+  grep "starting GCE discovery service" /tmp/elasticsearch_bootstrap_start.log && break;
+  sleep 10
+done
+
+sleep 10
+
+master_settings_missing=false
+wait_iterator=0
+
+echo "checking if master selection worked correctly"
+while true; do
+  if grep "master not discovered yet, this node has not previously joined a bootstrapped" /tmp/elasticsearch_bootstrap_start.log; then
+    master_settings_missing=true
+    break
+  fi
+  wait_iterator=$(($wait_iterator+1))
+  if [[ ${wait_iterator} -eq 10 ]]; then
+    break
+  fi
+  sleep 10
+done
+
+while pgrep java; do
+  pkill java
+  sleep 10
+done
+
+if [[ ${master_settings_missing} == "true" ]]; then
+
+  source /home/devops/master_list.sh
+
+  su elasticsearch -s /bin/bash -c "/usr/share/elasticsearch/bin/elasticsearch -Ecluster.initial_master_nodes=${master_list} | tee /tmp/elasticsearch_bootstrap_master_list_start.log >/dev/full" &
+
+  while ! grep "starting GCE discovery service" /tmp/elasticsearch_bootstrap_master_list_start.log; do
+    sleep 10
+  done
+
+  sleep 60
+  while pgrep java; do
+    pkill java
+    sleep 10
+  done
+fi
+
+sleep 60
+
 systemctl enable elasticsearch.service
+
+if [[ -f /tmp/elasticsearch_bootstrap_start.log ]]; then
+  rm /tmp/elasticsearch_bootstrap_start.log
+fi
+if [[ -f /tmp/elasticsearch_bootstrap_master_list_start.log ]]; then
+  rm /tmp/elasticsearch_bootstrap_master_list_start.log
+fi
```

Comment:
Bez toho, abych to celé důkladně zkoumala těm změnám v `bootstrap.sh` moc nerozumím.

Reply:
Tohle není zásadní, to je jen poklizení souborů, kam se ukládali logy pro grep

Reply:
To byl komentář k celému bootstrap.sh 🙂